DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala

```diff
@@ -563,7 +565,40 @@ object CommonProcessorFactory {
         batchLog.warn(s"End batch ${batchTime}, output partition time:${outputPartitionTime}, namespace:${namespace}")
 
         metrics ++ result
-      } // end of processPaths
+      }, // end of processPaths
+      /*
+      process a batch of ConsumerRecords from kafka
+     */
+      processConsumerRecord = (rdd: RDD[ConsumerRecord[String,String]], batchTime: Timestamp, batchInterval: Duration, outputPartitionTime: Timestamp) =>{
+        processDataset(rdd
+          .map(d=>{
+            val value = d.value()
+            if(value==null) throw new EngineException(s"null bytes from ConsumerRecord")
+            //Capture key if present. Key can be null.
+            val key = if(d.key!=null) Some("key"->d.key.toString) else None
+            (
+              value,
+              Map.empty[String, String],// Properties
+              Map[String, String](
+                "offset"->d.offset().toString,
+                "partition"->d.partition().toString,
+                "serializedKeySize"->d.serializedKeySize().toString,
+                "serializedValueSize"->d.serializedValueSize().toString,
+                "timestamp"->d.timestamp().toString,
+                "timestampType"->d.timestampType().toString,
+                "topic"->d.topic()
+              ) ++ key,
+              FileInternal())
+          })
+          .toDF(
+            ColumnName.RawObjectColumn,
+            "Properties",
+            "SystemProperties",
```

Comment:
Since these two constants are used in multiple places, let's add them to the constants list.

Reply:
Updated